### ISSUE TITLE
Plotting of functiongraph element shows line at discontinuity

### ISSUE THREAD

**Issue by StephanvandenBerkmortel** [jsxgraph/jsxgraph#587]
When using the `functiongraph` element, there are certain conditions where functions that contain a discontinuity within the domain still have a curve drawn at the discontinuity. In fiddle https://jsfiddle.net/qht8yzsc/1/ an example is provided of the behavior using the function $\sqrt{x^2 - 16}$. The problem seems to be dependent on the visible region, zooming in or out creates zoom levels that show the discontinuity correctly, which suggests it might be related to #245.

Additionally, I have only encountered the bug using the default `plotVersion: 2`; when switching to `plotVersion: 4` the problem seems to be resolved. Is there some literature available on the plotting algorithm used for version 2 to help investigate the bug further? Will the `plotVersion: 2` become obsolete in favor of version 4 at some point?

Kind regards

**Comment by alfredwassermann** [comment]
Dear Stephan,
thanks for pointing out this bug. Indeed, one special case has not been considered in the algorithm. This is hopefully fixed now.

plotVersion:2 is my own development, there is no literature about it. Actually, it tries to plot curves, not just function graphs, which makes the problem a little bit more complex. Its a recursive algorithm which consists of studying the relation of three consecutive points and try to detect jumps, cusps and non-smooth points from the angles in the triangle defined by the points.

plotVersion:4 is a promising study of an idea mentioned in Richard Fatemen (Honest plotting, global extrema, and interval arithmetic). It uses a fascinating difference method to detect critical points which works astonishingly good. However, so far there was no time to make it ready for production. At the time being, it is mainly working for graphing functions, not for general curves. 
The implementation is already excellent in detecting critical points, but the graphing of the area close to the critical point still needs work. Another weakness of the algorithm is that it has difficulties to handle highly periodic functions.

In summary, it will take while until v2 will become obsolete...
